public/app/features/plugins/admin/hooks/usePluginDetailsTabs.tsx

```diff
@@ -9,12 +9,15 @@ type ReturnType = {
   error: Error | undefined;
   loading: boolean;
   tabs: PluginDetailsTab[];
+  defaultTab: MutableRefObject<string>;
 };
 
 export const usePluginDetailsTabs = (plugin?: CatalogPlugin, defaultTabs: PluginDetailsTab[] = []): ReturnType => {
   const { loading, error, value: pluginConfig } = usePluginConfig(plugin);
   const isPublished = Boolean(plugin?.isPublished);
   const { pathname } = useLocation();
+  let defaultTab = useRef('');
```

Comment:
**Question:** maybe a dumb one, but did you use `useRef()` here to not cause rerenders?

Reply:
I used `useRef` due to defining the variable outside the useMemo the assignment will be lost after each render. It's not really necessary and looking at it again it's harder to reason with. Have moved it inside the useMemo and swapped the `ref` for a `let`. 👍

Reply:
Thanks for the explanation mate 🙏